consensus/istanbul/backend/announce.go

```diff
@@ -510,29 +608,29 @@ func (sb *Backend) generateAndGossipQueryEnode(version uint, enforceRetryBackoff
 			return nil, err
 		}
 
-		if err = sb.Gossip(payload, istanbul.QueryEnodeMsg); err != nil {
+		if err = m.support.Gossip(payload, istanbul.QueryEnodeMsg); err != nil {
 			return nil, err
 		}
 
-		if err = sb.valEnodeTable.UpdateQueryEnodeStats(valEnodeEntries); err != nil {
+		if err = m.valEnodeTable.UpdateQueryEnodeStats(valEnodeEntries); err != nil {
 			return nil, err
 		}
 	}
 
 	return qeMsg, err
 }
 
-func (sb *Backend) getQueryEnodeValEnodeEntries(enforceRetryBackoff bool) ([]*istanbul.AddressEntry, error) {
-	logger := sb.logger.New("func", "getQueryEnodeValEnodeEntries")
-	valEnodeEntries, err := sb.valEnodeTable.GetValEnodes(nil)
+func (m *AnnounceManager) getQueryEnodeValEnodeEntries(enforceRetryBackoff bool) ([]*istanbul.AddressEntry, error) {
+	logger := m.logger.New("func", "getQueryEnodeValEnodeEntries")
+	valEnodeEntries, err := m.valEnodeTable.GetValEnodes(nil)
 	if err != nil {
 		return nil, err
 	}
 
 	var queryEnodeValEnodeEntries []*istanbul.AddressEntry
 	for address, valEnodeEntry := range valEnodeEntries {
 		// Don't generate an announce record for ourselves
-		if address == sb.Address() {
+		if address == m.wallets().Ecdsa.Address {
```

Comment:
i realie that `wallets` is a strange name. Implies we have many wallets, as in many identities/accounts. But it's just that have an ecdsa key and a bls key. We should find a better name for this

Reply:
We have a discussion about that name in the exact same object for Backend with @piersy , but we couldn't find any. Definitely open for discussion but I would change that in a separate PR since it's already 'wallets' in Backend in master